### DISCUSSION TITLE
How do I start a tray icon?

### DISCUSSION THREAD

**Discussion by laduke** [tauri-apps/tauri#11567] [Q&A]
😅 
I have a preact vite hello world app working (on macos arm). 

I'm here https://tauri.app/plugin/system-tray/ looking at the javascript examples. I'm  not sure where to paste these snippets or if there is another start() method to call somewhere. But so far nothing is showing up in my menubar. 

edit:

using the rust api works if you set the .icon()

**Comment by Blushyes** [comment]
You can copy it until your software starts, just create a tray instance
![image](https://github.com/user-attachments/assets/361f0b52-779d-4bdd-96b8-3a21fb2b21ba)

**Comment by zhanglun** [comment]
Here's how I did it, you might find it helpful to refer to:

1. Modify the `bundle.resources` in `tauri.config.json`:

```
...
"bundle": {
    "resources": [
      "icons/**/*", 
    ],
}
...
```
2. Load the image:
```rust
let icon_path = app.path().resolve("icons/tray.png", tauri::path::BaseDirectory::Resource)?;

let tray_icon = TrayIconBuilder::new()
    .icon_as_template(true)
    .icon(tauri::image::Image::from_path(icon_path).unwrap());
```
Here's a more detailed explanation:

Modify the `tauri.config.json` file to include your icon resources:

In your `tauri.config.json` file, locate the bundle section and add the resources array if it doesn't exist. Inside this array, include a reference to your icon files. The path "icons/**/*" tells Tauri to include all files within the icons directory. 
Load the image for the tray icon in Rust:

First, determine the path to your tray icon image. This is done by resolving the path to "icons/tray.png" relative to the application's resource directory `src-tauri/icons`.
Then, create a new `TrayIconBuilder` instance. Set icon_as_template to true if you want the icon to be used as a template, which can be styled by the system.
Finally, load the image from the specified path and assign it to the tray icon builder using the icon method. The unwrap method is used to handle the result, assuming the image path is correct and the image can be loaded without errors.
This approach ensures that your tray icon is properly set up and displayed in your Tauri application.
